napari/_qt/widgets/qt_highlight_preview.py

```diff
@@ -405,52 +426,81 @@ def __init__(
         ----------
-        value : int
-            Highlight value.
+        thickness_value : int
+            Highlight thickness value.
         """
-        if value == '':
+        if thickness_value == '':
             return
-        value = int(value)
-        value = max(min(value, self._max_value), self._min_value)
-        if value == self._value:
+        thickness_value = int(thickness_value)
+        thickness_value = max(
+            min(thickness_value, self._max_value), self._min_value
+        )
+        if thickness_value == self._thickness_value:
             return
-        self._value = value
+        self._thickness_value = thickness_value
+        self._value['highlight_thickness'] = self._thickness_value
+        self.valueChanged.emit(self._value)
+        self._refresh()
+
+    def _update_color_value(self, color_value):
+        """Update highlight color value.
+
+        Parameters
+        ----------
+        color_value : List[float]
+            Highlight color value as a list of floats.
+        """
+        if isinstance(color_value, np.ndarray):
+            color_value = color_value.tolist()
+        if color_value == self._color_value:
+            return
+        self._color_value = color_value
+        self._value['highlight_color'] = self._color_value
         self.valueChanged.emit(self._value)
         self._refresh()
 
     def _refresh(self):
         """Set every widget value to the new set value."""
         self.blockSignals(True)
-        self._lineedit.setText(str(self._value))
-        self._slider.setValue(self._value)
-        self._triangle.setValue(self._value)
-        self._preview.setValue(self._value)
+        color = QColor(
+            *(
+                np.ceil(
+                    np.array(self._color_value) * [255, 255, 255, 255]
+                ).astype(int)[:-1]
+            )
+        )
+        self._lineedit.setText(str(self._thickness_value))
+        self._slider.setValue(self._thickness_value)
+        self._triangle.setValue(self._thickness_value, color=color)
+        self._preview.setValue(self._thickness_value, color=color)
+        self._color_swatch_edit.setColor(self._color_value)
         self.blockSignals(False)
 
     def value(self):
         """Return current value.
 
         Returns
         -------
-        int
-            Current value of highlight widget.
+        dict
+            Current value of highlight widget (thickness and color).
         """
         return self._value
 
-    def setValue(self, value):
+    def setValue(self, value: dict):
         """Set new value and update widget.
 
         Parameters
         ----------
-        value : int
-            Highlight value.
+        value : dict
+            Highlight value (thickness and color).
         """
-        self._update_value(value)
+        self._update_thickness_value(dict(value)['highlight_thickness'])
+        self._update_color_value(dict(value)['highlight_color'])
```

Comment:
why the `dict` cast?

Reply:
It looks like bad type annotation..

Reply:
I was getting an error message: `TypeError: 'HighlightSettings' object is not subscriptable`, which the cast made disappear. However, giving this a more in deep check I think the error was caused due to the event connections I added in the appearance settings code (`self.highlight.events.connect(self._highlight_changed)`). I added that following the approach for the `DaskSettings` https://github.com/napari/napari/blob/507c4688f38873ffd532034ea289cb3fa8562934/napari/settings/_application.py#L24-L40 but maybe that connection is not necessary here? 🤔

Also, tested and if I remove that logic I'm able to remove the `dict` cast 🤔 For the moment seems like I can remove that manual event connection, right? Anyhow, any info regarding the need for it is appreciated :)